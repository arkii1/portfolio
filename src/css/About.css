
@media screen and (max-width: 499px) {
    .about__flex {
        flex-direction: column;
        align-items: center;
    }

    .about__details {
        justify-content: flex-start;
    }

    .about__grid__item__icon {
        width: min(80px, calc(30vw - 1rem));
    }
    
    .about__details > p {
        align-self: center;
        text-align: center;
    }

    .about__img {
        width: calc(100vw - 3rem);
    }
}

@media screen and (min-width: 499px) and (max-width: 999px) {
    .about__flex {
        flex-direction: column;
        align-items: center;
    }

    .about__details {
        justify-content: flex-start;
    }

    .about__grid__item__icon {
        width: 140px;
    }
    
    .about__details > p {
        align-self: center;
        text-align: center;
    }

    .about__img {
        width: min(600px, calc(100vw - 3rem));
    }
}

@media screen and (min-width: 1000px) {
    .about__flex {
        flex-direction: row;
    }

    .about__details {
        justify-content: center;
    }

    .about__grid {
        gap: 1rem;
    }

    .about__grid__item__icon {
        width: min(10vw, 160px);
    }

    .about__img {
        width: min(30vw, 400px);
    }

    .about__details > p {
        align-self: left;
    }

}

.about {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 1rem;
    padding: 0 1rem;
    word-wrap: break-word;
}

.about__flex {
    display: flex;
    gap: 1rem;
}

.about__header {
    font-size: 2.2rem;
    align-self: center;
}

.about__details {
    display: flex;
    flex-direction: column;
    gap: 1rem;
}

.about__details > p {
    max-width: 30rem;
}

.about__grid {
    display: grid;
    padding: 1rem 0;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
}

.about__grid__item {
    list-style: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.about__grid__item > span {
    font-family: 'Source Code Pro';
    letter-spacing: 0.1rem;
}