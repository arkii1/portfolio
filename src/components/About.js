import React from "react"
import "../css/About.css"

import htmlLogo from "../Assets/images/htmlLogo.svg"
import cssLogo from "../Assets/images/cssLogo.svg"
import jsLogo from "../Assets/images/jsLogo.svg"
import reactLogo from "../Assets/images/reactLogo.svg"
import gitLogo from "../Assets/images/gitLogo.svg"
import webpackLogo from "../Assets/images/webpackLogo.svg"
import npmLogo from "../Assets/images/npmLogo.svg"
import reduxLogo from "../Assets/images/reduxLogo.svg"
import jestLogo from "../Assets/images/jestLogo.svg"

function About() {
  return (
    <div className="about">
      <h2 className="about__header">About</h2>
      <hr className="break" />
      <div className="about__flex">
        <div className="about__details">
          <img
            className="about__img"
            src="https://freesvg.org/img/abstract-user-flat-4.png"
            alt=""
          />
          <p>
            A Mathematics graduate with a passion for creating, technology and
            learning.
          </p>
          <p>
            During my degree in 2018, I started learning game development and
            design in my spare time using Unity and C#. This gave me all the
            traits necessary to succeed when I decided to make the switch to
            full stack web development.
          </p>
          <p>
            I&apos;m mainly interested in working with Javascript using React,
            NodeJS and Express, however I also have expeirience with C#, C++,
            Python and R.
          </p>
        </div>
        <div className="about__grid">
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={htmlLogo} alt="" />
            <span className="about__grid__item__name">HTML5</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={cssLogo} alt="" />
            <span className="about__grid__item__name">CSS3</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={jsLogo} alt="" />
            <span className="about__grid__item__name">Javascript</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={reactLogo} alt="" />
            <span className="about__grid__item__name">React</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={gitLogo} alt="" />
            <span className="about__grid__item__name">Git</span>
          </li>

          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={webpackLogo} alt="" />
            <span className="about__grid__item__name">Webpack</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={reduxLogo} alt="" />
            <span className="about__grid__item__name">Redux</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={npmLogo} alt="" />
            <span className="about__grid__item__name">npm</span>
          </li>
          <li className="about__grid__item">
            <img className="about__grid__item__icon" src={jestLogo} alt="" />
            <span className="about__grid__item__name">Jest</span>
          </li>
        </div>
      </div>
    </div>
  )
}

export default About
