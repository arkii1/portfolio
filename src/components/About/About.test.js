import React from "react"
import ReactDOM from "react-dom/client"
import { cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import About from "./About"

afterEach(cleanup)

it("renders without crashing", () => {
  const container = document.createElement("div")
  const root = ReactDOM.createRoot(container)
  root.render(<About />)
})

it("matches snapshot", () => {
  const tree = renderer.create(<About />).toJSON()
  expect(tree).toMatchSnapshot()
})
