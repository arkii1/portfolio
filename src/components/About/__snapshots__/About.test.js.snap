// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="about"
>
  <h2
    className="about__header"
  >
    About
  </h2>
  <hr
    className="break"
  />
  <div
    className="about__flex"
  >
    <div
      className="about__details"
    >
      <img
        alt=""
        className="about__img"
        src="https://freesvg.org/img/abstract-user-flat-4.png"
      />
      <p>
        A Mathematics graduate with a passion for creating, technology and learning.
      </p>
      <p>
        During my degree in 2018, I started learning game development and design in my spare time using Unity and C#. This gave me all the traits necessary to succeed when I decided to make the switch to full stack web development.
      </p>
      <p>
        I'm mainly interested in working with Javascript using React, NodeJS and Express, however I also have expeirience with C#, C++, Python and R.
      </p>
    </div>
    <div
      className="about__grid"
    >
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="htmlLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          HTML5
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="cssLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          CSS3
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="jsLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          Javascript
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="reactLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          React
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="gitLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          Git
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="webpackLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          Webpack
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="reduxLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          Redux
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="npmLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          npm
        </span>
      </li>
      <li
        className="about__grid__item"
      >
        <img
          alt=""
          className="about__grid__item__icon"
          src="jestLogo.svg"
        />
        <span
          className="about__grid__item__name"
        >
          Jest
        </span>
      </li>
    </div>
  </div>
</div>
`;
